### PR TITLE
NEAP v2 resource

### PR DESCRIPTION
**Pull request by dmitrykruglov**

### CODE REVIEW

File: examples/resources/itsi_notable_event_aggregation_policy/resource.tf [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        limit    = 1
+        operator = "=="
+      }
+    }
+  }
+  rule {
+    actions {
+      item {
+        custom {
+          type = "email"
```
Comment (masha1997) [inline examples/resources/itsi_notable_event_aggregation_policy/resource.tf:49]: nice example, thank you for showing custom actions as well

File: provider/resource_neap.go [hunk 2]
```diff
@@ -0,0 +1,1467 @@
...
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/tivo/terraform-provider-splunk-itsi/models"
+	"github.com/tivo/terraform-provider-splunk-itsi/provider/util"
+)
+
+/*
+	NEAP Resource
+	Table of contents:
+		(1) [ NEAP data structures, constants and variables ]
```
Comment (masha1997) [inline provider/resource_neap.go:38]: nice, it brings much more readability in long implementation, I should do the same

File: provider/resource_neap.go [hunk 3]
```diff
@@ -0,0 +1,1467 @@
...
+	}, diags
+}
+
+// (5) [ Neap Parse Workflow ]__________________________________________________
+
+type neapParseWorkflow struct{}
+
+var _ apiparseWorkflow[neapModel] = &neapParseWorkflow{}
+
+//lint:ignore U1000 used by apiparser
```
Comment (masha1997) [inline provider/resource_neap.go:1219]: I didn't understand that comment :) I read about linter a bit, didn't use this, will set it up in vscode as well.

**Review by masha1997 — APPROVED** [review]
This is VERY neat and cool, thank you for taking care of this!

**Comment by masha1997** [comment]
Just a question, how the plan look like when for example the action item in the rule updates?

File: provider/resource_neap.go (hunk 2 above)
Comment (dmitrykruglov) [reply provider/resource_neap.go:38]: yeah i think it helps a bit with navigating this file and keeping things more organized :)

File: provider/resource_neap.go (hunk 3 above)
Comment (dmitrykruglov) [reply provider/resource_neap.go:1219]: This comment tells staticcheck (go linter) to ignore the warning that it gives about this function not being used.
Screenshot 2024-04-11 at 8 19 37 AM

Comment (dmitrykruglov) [reply provider/resource_neap.go:1219]: I think the warning was due to the fact that the function is implementing an interface, as isn't called directly as *neapBuildWorkflow's method, so staticcheck couldn't figure out it's actually being used. So i added the comment for now to ignore that.

Comment (dmitrykruglov) [reply provider/resource_neap.go:1219]: But yeah, you should use the linter, it actually comes with the standard official VSCode's "Go" extension. I think you just need to run "Go Install/Update Tools" in the VS Code's Command Palette 
Screenshot 2024-04-11 at 8 49 09 AM

**Comment by dmitrykruglov** [comment]
> Just a question, how the plan look like when for example the action item in the rule updates?

it would like the entire parent `actions` block is changing, because it's a set i think.. I guess i can check if making it a list would produce a better plan, but probably not a huge deal..
```
  # itsi_notable_event_aggregation_policy.neap will be updated in-place
  ~ resource "itsi_notable_event_aggregation_policy" "neap" {
        id                          = "5c50c683-109a-eb9e-4635-167c3efec6cf"
        # (15 unchanged attributes hidden)

      ~ rule {
            id       = "21e745bc-bfc3-f9b0-0e18-a1a3da2dc87a"
            # (1 unchanged attribute hidden)

          - actions {
              - condition = "AND" -> null

              - item {
                  - change_severity = "medium" -> null
                  - execute_on      = "GROUP" -> null
                }
              - item {
                  - change_status = "in progress" -> null
                  - execute_on    = "GROUP" -> null
                }
              - item {
                  - execute_on = "GROUP" -> null

                  - custom {
                      - config = jsonencode(
                            {
                              - bcc          = ""
                              - cc           = ""
                              - content_type = "html"
                              - message      = <<-EOT
                                    Status $result.slack_footer_msg$

                                    $result.email_msg$ Dashboard: $result.dashboard$
                                    Runbook: $result.runbook$

                                    View in Splunk ITSI
                                EOT
                              - priority     = "normal"
                              - subject      = "$result.slack_header$"
                              - to           = "$result.email$"
                            }
                        ) -> null
                      - type   = "email" -> null
                    }
                }
            }
          + actions {
              + condition = "AND"

              + item {
                  + change_severity = "medium"
                  + execute_on      = "GROUP"
                }
              + item {
                  + change_status = "closed"
                  + execute_on    = "GROUP"
                }
              + item {
                  + execute_on = "GROUP"

                  + custom {
                      + config = jsonencode(
                            {
                              + bcc          = ""
                              + cc           = ""
                              + content_type = "html"
                              + message      = <<-EOT
                                    Status $result.slack_footer_msg$

                                    $result.email_msg$ Dashboard: $result.dashboard$
                                    Runbook: $result.runbook$

                                    View in Splunk ITSI
                                EOT
                              + priority     = "normal"
                              + subject      = "$result.slack_header$"
                              + to           = "$result.email$"
                            }
                        )
                      + type   = "email"
                    }
                }
            }

            # (1 unchanged block hidden)
        }

        # (4 unchanged blocks hidden)
    }
```

**Comment by masha1997** [comment]
much better than previously, so only one rule in plan, thank you

**Pull request merged**